File: tests/java/TestStrategy.java
public class TestStrategy {
    public static void main(String[] args) {
        System.out.println("Testing Java Strategy...");
        try {
            client.main(args);
        } catch (Exception e) {
            e.printStackTrace();
            System.exit(1);
        }
        System.out.println("Java Strategy Test Passed!");
    }
}

File: tests/c/TestStrategy.c
#define main pattern_main
#include "../../oop_in_c/Strategy/Strategy.c"
#undef main
#include <stdio.h>

int main() {
    printf("Testing C Strategy...\n");
    // Call pattern main
    pattern_main();
    printf("C Strategy Test Passed!\n");
    return 0;
}
